src/rpp/rpp/schedulers/trampoline_scheduler.hpp

```diff
@@ -0,0 +1,92 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2022 - present.
+//                            TC Wang 2022 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/schedulers/fwd.hpp>                       // own forwarding
+#include <rpp/schedulers/details/worker.hpp>            // worker
+#include <rpp/subscriptions/composite_subscription.hpp> // lifetime
+#include <rpp/schedulers/details/queue_worker_state.hpp>// state
+
+#include <concepts>
+#include <chrono>
+#include <functional>
+#include <thread>
+
+thread_local bool queue_in_use = false;
+thread_local rpp::schedulers::details::queue_worker_state queue{};
+
+namespace rpp::schedulers
+{
+
+template <typename F>
+struct scope_guard {
+    explicit scope_guard(F&& f) : m_f(std::forward<F>(f)) {}
+    ~scope_guard() noexcept { m_f(); }
+    F m_f;
+};
+
+/**
+ * \brief schedules execution of schedulables via queueing tasks to the caller thread with priority to time_point and order
+ *
+ * \par Example
+ * \snippet trampoline.cpp trampoline
+ *
+ * \ingroup schedulers
+ */
+class trampoline final : public details::scheduler_tag
+{
+public:
+    class worker_strategy
+    {
+    public:
+        explicit worker_strategy(const rpp::composite_subscription& subscription)
+            : m_sub{subscription} {};
+
+        void defer_at(time_point time_point, std::invocable auto&& fn) const
+        {
+            if (!m_sub.is_subscribed())
+                return;
+
+            queue.emplace(time_point, std::forward<decltype(fn)>(fn));
+            if (queue_in_use)
+                return;
+
+            queue_in_use = true;
+            scope_guard cleanup([]() noexcept
+            {
+                // If error occurs, we still want the thread-local state to be reset.
+                queue_in_use = false;
+            });
+            while (!queue.is_empty() && m_sub.is_subscribed())
+            {
+                std::function<void()> schedulable{};
+                std::stop_token stop;
```

Comment:
stop token used to unblock wait in case of someone requested stop. Add it to `m_sub` to request stop in case of unsubscribe

Reply:
Yup, I have the same question too. I'll add that later.